src/Arcus.BackgroundJobs.Tests.Integration/AzureActiveDirectory/ClientSecretExpirationJobTests.cs

```diff
@@ -39,49 +39,65 @@ public ClientSecretExpirationJobTests(ITestOutputHelper outputWriter)
         public async Task CheckExpiredOrAboutToExpireClientSecretsInAzureActiveDirectory_PublishEvents()
         {
             // Arrange
-            var topicEndpoint = _config.GetValue<string>("Arcus:Infra:EventGrid:TopicUri");
-            var topicEndpointSecret = _config.GetValue<string>("Arcus:Infra:EventGrid:AuthKey");
-
-            IEventGridPublisher eventGridPublisher = EventGridPublisherBuilder
-                .ForTopic(topicEndpoint)
-                .UsingAuthenticationKey(topicEndpointSecret)
-                .Build();
-
+            int expirationThreshold = 14;
+            IEventGridPublisher eventGridPublisher = CreateEventGridPublisher();
             var options = new WorkerOptions();
             options.ConfigureLogging(_logger)
                    .AddSingleton<IEventGridPublisher>(eventGridPublisher)
-                   .AddClientSecretExpirationJob(
-                        options => {
-                            options.RunImmediately = true;
-                            options.ExpirationThreshold = 14;
-                            options.EventUri = new Uri("https://github.com/arcus-azure/arcus.backgroundjobs");
-                        });
+                   .AddClientSecretExpirationJob(opt => 
+                   {
+                       opt.RunImmediately = true;
+                       opt.ExpirationThreshold = expirationThreshold;
+                       opt.EventUri = new Uri("https://github.com/arcus-azure/arcus.backgroundjobs");
+                   });
+
+            var tenantId = _config.GetValue<string>("Arcus:AzureActiveDirectory:TenantId");
+            var clientId = _config.GetValue<string>("Arcus:AzureActiveDirectory:ServicePrincipal:ClientId");
+            var clientSecret = _config.GetValue<string>("Arcus:AzureActiveDirectory:ServicePrincipal:ClientSecret");
```

Comment:
I'll give you the secrets to test locally, if you want. I already updated Azure DevOps so it will run remotely.

Reply:
That would be very helpful, thanks!

Reply:
Shared. 👍